### PR TITLE
Terapia,Notifica,TutoreRest

### PR DESCRIPTION
**Pull request by matteoaldi**

### CODE REVIEW

File: src/main/java/com/glucoseguardian/webbackend/application/rest/NotificaRest.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+/**
+ * implementation NotificaRest.
+ */
+public class NotificaRest {
```
Comment (handymenny) [inline src/main/java/com/glucoseguardian/webbackend/application/rest/NotificaRest.java]: Missing RestController Annotation, example:
@RestController
@RequestMapping("notifica")

File: src/main/java/com/glucoseguardian/webbackend/application/rest/NotificaRest.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      ex.printStackTrace();
+    }
+    return CompletableFuture.completedFuture(response);
+  }
+
+  /**
+   * Metodo che gestisce il servizio save Notifica.
+   */
+
+  @PostMapping(value = "/save", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (handymenny) [inline src/main/java/com/glucoseguardian/webbackend/application/rest/NotificaRest.java]: Wrong endpoint, this should be "/send", e.g.:
@PostMapping(value = "/send", produces = MediaType.APPLICATION_JSON_VALUE)

File: src/main/java/com/glucoseguardian/webbackend/application/rest/NotificaRest.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    if (result) {
+      return CompletableFuture.completedFuture(
+          new ResponseEntity<>(new RisultatoDto("Notifica inserito correttamente"), HttpStatus.OK));
+    } else {
+      return CompletableFuture.completedFuture(
+          new ResponseEntity<>(new RisultatoDto("Errore durante l'inserimento della Notifica"),
+              HttpStatus.INTERNAL_SERVER_ERROR));
+    }
+  }
+
```
Comment (handymenny) [inline src/main/java/com/glucoseguardian/webbackend/application/rest/NotificaRest.java]: Missing updateStato endpoint. This method is missing in Service too.. So it should be added there first.

File: src/main/java/com/glucoseguardian/webbackend/application/rest/TerapiaRest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+          HttpStatus.INTERNAL_SERVER_ERROR);
+      ex.printStackTrace();
+    }
+    return CompletableFuture.completedFuture(response);
+  }
+
+  /**
+   * Metodo che gestisce il servizio save Terapia.
+   */
+
```
Comment (handymenny) [inline src/main/java/com/glucoseguardian/webbackend/application/rest/TerapiaRest.java]: Missing update endpoint

**Review by AmineSr97 — APPROVED** [review]

**Review by handymenny — APPROVED** [review]

**Pull request merged**